Next package.json does not include all required dependencies

## Expected Behavior

I create a Next app. This app uses `styled-components` for styling. I run `nx build myapp` build the app. I copy from `dist/apps/myapp` to a separate directory to test that the app is ready to be deployed. When I test the app by running `npm start`, I can run the app by going to `localhost:3000`. 

## Current Behavior

I get an error: `Error: Cannot find module 'styled-components'` when going to `localhost:3000`. 

When running the app locally using `nx serve myapp`, everything works as expected. 

## Failure Information (for bugs)

Looking at the generated `package.json`, it looks like `styled-components` are a missing dependency. 

```
{
  "name": "myapp",
  "version": "0.0.1",
  "scripts": {
    "start": "next start"
  },
  "dependencies": {
    "next": "9.3.3",
    "react": "16.12.0",
    "react-dom": "16.12.0"
  },
  "devDependencies": {}
}
```

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Generate a Next app that uses styled-components.
2. Build the app.
3. Move the built app to a different directory and run it. 

### Context

Please provide any relevant information about your setup:

```
  @nrwl/angular : Not Found
  @nrwl/cli : 9.3.0
  @nrwl/cypress : 9.3.0
  @nrwl/eslint-plugin-nx : 9.3.0
  @nrwl/express : Not Found
  @nrwl/jest : 9.3.0
  @nrwl/linter : 9.3.0
  @nrwl/nest : 9.3.0
  @nrwl/next : 9.3.0
  @nrwl/node : 9.3.0
  @nrwl/react : 9.3.0
  @nrwl/schematics : Not Found
  @nrwl/tao : 9.3.0
  @nrwl/web : 9.3.0
  @nrwl/workspace : 9.3.0
  typescript : 3.8.3
```
`"styled-components": "5.0.1"`

### Failure Logs

```
Error: Cannot find module 'styled-components'
Require stack:
- /Users/olya777/Projects/myapp/.next/server/static/_qz_WuDsra8FGG8QTZZg3/pages/_document.js
- /Users/olya777/Projects/myapp/node_modules/next/dist/next-server/server/load-components.js
- /Users/olya777/Projects/myapp/node_modules/next/dist/next-server/server/api-utils.js
- /Users/olya777/Projects/myapp/node_modules/next/dist/next-server/server/next-server.js
- /Users/olya777/Projects/myapp/node_modules/next/dist/server/next.js
- /Users/olya777/Projects/myapp/node_modules/next/dist/server/lib/start-server.js
- /Users/olya777/Projects/myapp/node_modules/next/dist/cli/next-start.js
- /Users/olya777/Projects/myapp/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:797:15)
    at Function.Module._load (internal/modules/cjs/loader.js:690:27)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.Dtiu (/Users/olya777/Projects/myapp/.next/server/static/_qz_WuDsra8FGG8QTZZg3/pages/_document.js:780:18)
    at __webpack_require__ (/Users/olya777/Projects/myapp/.next/server/static/_qz_WuDsra8FGG8QTZZg3/pages/_document.js:23:31)
    at Module.Y7Z2 (/Users/olya777/Projects/myapp/.next/server/static/_qz_WuDsra8FGG8QTZZg3/pages/_document.js:917:75)
    at __webpack_require__ (/Users/olya777/Projects/myapp/.next/server/static/_qz_WuDsra8FGG8QTZZg3/pages/_document.js:23:31)
    at Object.1 (/Users/olya777/Projects/myapp/.next/server/static/_qz_WuDsra8FGG8QTZZg3/pages/_document.js:99:18)
    at __webpack_require__ (/Users/olya777/Projects/myapp/.next/server/static/_qz_WuDsra8FGG8QTZZg3/pages/_document.js:23:31) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/olya777/Projects/myapp/.next/server/static/_qz_WuDsra8FGG8QTZZg3/pages/_document.js',
    '/Users/olya777/Projects/myapp/node_modules/next/dist/next-server/server/load-components.js',
    '/Users/olya777/Projects/myapp/node_modules/next/dist/next-server/server/api-utils.js',
    '/Users/olya777/Projects/myapp/node_modules/next/dist/next-server/server/next-server.js',
    '/Users/olya777/Projects/myapp/node_modules/next/dist/server/next.js',
    '/Users/olya777/Projects/myapp/node_modules/next/dist/server/lib/start-server.js',
    '/Users/olya777/Projects/myapp/node_modules/next/dist/cli/next-start.js',
    '/Users/olya777/Projects/myapp/node_modules/next/dist/bin/next'
  ]
}
```

--- a/packages/next/src/builders/build/lib/create-package-json.ts
+++ b/packages/next/src/builders/build/lib/create-package-json.ts
@@ -3,7 +3,6 @@ import { writeFileSync } from 'fs';
 import { join } from 'path';
 import { readJsonFile } from '@nrwl/workspace';
 import { NextBuildBuilderOptions } from '../../../utils/types';
-import { reactDomVersion, reactVersion } from '@nrwl/react/src/utils/versions';
 
 export async function createPackageJson(
   options: NextBuildBuilderOptions,
@@ -13,23 +12,13 @@ export async function createPackageJson(
     join(context.workspaceRoot, 'package.json')
   );
 
-  const allWorkspaceDeps = {
-    ...rootPackageJson.dependencies,
-    ...rootPackageJson.devDependencies,
-  };
-
   const outPackageJson = {
     name: context.target.project,
     version: '0.0.1',
     scripts: {
       start: 'next start',
     },
-    dependencies: {
-      next: allWorkspaceDeps.next,
-      react: allWorkspaceDeps.react || reactVersion,
-      'react-dom': allWorkspaceDeps['react-dom'] || reactDomVersion,
-    },
-    devDependencies: {},
+    dependencies: rootPackageJson.dependencies,
   };
 
   writeFileSync(

